contracts/non-fungible-tokens/andromeda-crowdfund/src/contract.rs

```diff
@@ -67,6 +58,28 @@ pub fn instantiate(
             owner: msg.owner,
         },
     )?;
+
+    if let Asset::Cw20Token(addr) = msg.campaign_config.denom.clone() {
+        let addr = addr.get_raw_address(&deps.as_ref())?;
+        ADOContract::default().permission_action(SEND_CW20_ACTION, deps.storage)?;
+        ADOContract::set_permission(
+            deps.storage,
+            SEND_CW20_ACTION,
+            addr,
+            Permission::Whitelisted(None),
+        )?;
+    }
+
+    let branch = DepsMut {
```

Comment:
Is this different than using `deps.branch()` ?

Reply:
Not really. branch() function requires `deps` to be  declared as mutable. To use that, `instantiate` function signature had to changed as follow (otherwise it throws an error).
```
pub fn instantiate(
    mut deps: DepsMut,
    env: Env,
    info: MessageInfo,
    msg: InstantiateMsg,
) -> Result<Response, ContractError> {
```

Reply:
Updated to use `deps.branch()`